### PR TITLE
Improvements to Guild command

### LINKED ISSUE #102: Improvements to Guild command
Some minor improvements need to be done to the guild command. These should be relatively easy.

---
Add a `;guild info` command that shows information such as:
- The flags that are enabled/disabled. This should be shown as a list of all the tags with a ✅ or a ❌next to it
- The date our bot joined the guild
 
Nothing else for now, this will eventually include more information such as whether the guild is blocked etc. This `;guild info` command should also have an optional guildId argument like so `;guild info [guildId]`

---
Add to the `;flag list` command by telling us what flags the user has enabled at the time. This could be done with ✅ and ❌ or some other way.

### PR DESCRIPTION
**Pull request by JamesHawkinss**
# Pull request

 - [x] I have read and agreed to the [code of conduct](https://github.com/CascadeBot/CascadeBot/blob/master/.github/CODE_OF_CONDUCT.md).
 - [x] I have read and complied with the [contributing guidelines](https://github.com/CascadeBot/CascadeBot/blob/master/.github/CONTRIBUTING.md).
 - [x] What I'm implementing was assigned to me or was not being worked on and is a somewhat requested feature. For reference see our [GitHub projects](https://github.com/orgs/CascadeBot/projects).
 - [x] I have tested all of my changes.
 
## Added/Changed
Following issue #102 I'm adding a ;guild list command

Closes #102

### CODE REVIEW

**Comment by binaryoverload** [comment]
You haven't done what the issue detailed. Please re-read issue #102

**Comment by JamesHawkinss** [comment]
That's why I said it's WIP, I just wanted to add what I had at the moment.

**Comment by binaryoverload** [comment]
Ah gotcha, please nake sure to open a draft PR in future

**Comment by JamesHawkinss** [comment]
Ready for review y'all

File: src/main/java/org/cascadebot/cascadebot/commands/subcommands/guild/GuildInfoSubCommand.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+public class GuildInfoSubCommand implements ICommandExecutable {
+
+    @Override
+    public void onCommand(Member sender, CommandContext context) {
+        GuildData dataForList = context.getData();
+        Guild guildForList = context.getGuild();
+
+        if (context.getArgs().length > 0) {
+            guildForList = CascadeBot.INS.getShardManager().getGuildById(context.getArg(0));
+            dataForList = GuildDataManager.getGuildData(Long.valueOf(context.getArg(0)));
```
Comment (binaryoverload) [inline src/main/java/org/cascadebot/cascadebot/commands/subcommands/guild/GuildInfoSubCommand.java]: We need to do more error handling here such as:
 - What if the arg isn't a number
 - What if the guild can't be found (It should be used like this `GuildDataManager.getGuildData(guildForList.getIdLong());`)

File: src/main/java/org/cascadebot/cascadebot/commands/subcommands/guild/GuildInfoSubCommand.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+
+        GuildData finalDataForList = dataForList;
+        String flags = Arrays.stream(Flag.values())
+                             .map(flag -> FormatUtils.formatEnum(flag) + " - " +
+                                     (finalDataForList.isFlagEnabled(flag) ? UnicodeConstants.TICK : UnicodeConstants.RED_CROSS))
+                             .collect(Collectors.joining("\n"));
+
+        builder.addField("Flags", flags, false);
+        builder.addField("Join Date", FormatUtils.formatDateTime(context.getSelfMember().getJoinDate()), false);
```
Comment (binaryoverload) [inline src/main/java/org/cascadebot/cascadebot/commands/subcommands/guild/GuildInfoSubCommand.java]: Perhaps we should add enabled modules as well just so the information is in one place?

**Comment by JamesHawkinss** [comment]
Added ;flags command: https://i.thedudeinthecorner.xyz/files/CZS65.png

File: src/main/java/org/cascadebot/cascadebot/commands/developer/GuildInfoSubCommand.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    @Override
+    public String command() {
+        return "info";
+    }
+
+    @Override
+    public CascadePermission getPermission() {
+        return null;
```
Comment (weeryan17) [inline src/main/java/org/cascadebot/cascadebot/commands/developer/GuildInfoSubCommand.java]: Add permissions to these.

File: src/main/java/org/cascadebot/cascadebot/commands/management/FlagsCommand.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+import org.cascadebot.cascadebot.data.objects.Flag;
+import org.cascadebot.cascadebot.data.objects.GuildData;
+import org.cascadebot.cascadebot.messaging.MessagingObjects;
+import org.cascadebot.cascadebot.permissions.CascadePermission;
+import org.cascadebot.cascadebot.utils.FormatUtils;
+
+import java.util.Arrays;
+import java.util.stream.Collectors;
+
+public class FlagsCommand implements ICommandMain {
```
Comment (weeryan17) [inline src/main/java/org/cascadebot/cascadebot/commands/management/FlagsCommand.java]: why do we need this command? If they donated to the bot they should now what they're getting. Flags only really serve as a back end for donation stuff

File: src/main/java/org/cascadebot/cascadebot/commands/developer/GuildInfoSubCommand.java [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+            return;
+        }
+
+        EmbedBuilder builder = MessagingObjects.getClearThreadLocalEmbedBuilder();
+        builder.setTitle(guildForList.getName());
+        builder.setThumbnail(guildForList.getIconUrl());
+
+
+        GuildData finalDataForList = dataForList;
+        String flags = Arrays.stream(Flag.values())
```
Comment (weeryan17) [inline src/main/java/org/cascadebot/cascadebot/commands/developer/GuildInfoSubCommand.java]: See review on flag command

File: src/main/java/org/cascadebot/cascadebot/commands/developer/GuildInfoSubCommand.java (hunk 3 above)
Comment (binaryoverload) [reply src/main/java/org/cascadebot/cascadebot/commands/developer/GuildInfoSubCommand.java]: These are Dev commands, they shouldn't have permissions

File: src/main/java/org/cascadebot/cascadebot/commands/management/FlagsCommand.java (hunk 4 above)
Comment (binaryoverload) [reply src/main/java/org/cascadebot/cascadebot/commands/management/FlagsCommand.java]: You're probably right. My thought behind it is that I would like some sort of way for admins to see what the server has

File: src/main/java/org/cascadebot/cascadebot/commands/developer/GuildInfoSubCommand.java (hunk 3 above)
Comment (weeryan17) [reply src/main/java/org/cascadebot/cascadebot/commands/developer/GuildInfoSubCommand.java]: Oh whops. i didn't realize this was a dev command.

File: src/main/java/org/cascadebot/cascadebot/commands/management/FlagsCommand.java (hunk 4 above)
Comment (weeryan17) [reply src/main/java/org/cascadebot/cascadebot/commands/management/FlagsCommand.java]: I mean i could see how that'd be nice, but we need to add like descriptions to them so users acctuly know what they do.

Comment (binaryoverload) [reply src/main/java/org/cascadebot/cascadebot/commands/management/FlagsCommand.java]: Agreed, I have a way I think I can do that nicely in Lang. Let's get this merged for now

Comment (weeryan17) [reply src/main/java/org/cascadebot/cascadebot/commands/management/FlagsCommand.java]: One thing first though

File: src/main/java/org/cascadebot/cascadebot/commands/management/FlagsCommand.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    @Override
+    public CascadePermission getPermission() {
+        return null;
+    }
+
+    @Override
+    public String description() {
+        return null;
```
Comment (weeryan17) [inline src/main/java/org/cascadebot/cascadebot/commands/management/FlagsCommand.java]: This isn't a developer command add permissions to this.

File: src/main/java/org/cascadebot/cascadebot/commands/management/FlagsCommand.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    }
+
+    @Override
+    public String command() {
+        return "flags";
+    }
+
+    @Override
+    public CascadePermission getPermission() {
+        return null;
```
Comment (weeryan17) [inline src/main/java/org/cascadebot/cascadebot/commands/management/FlagsCommand.java]: This isn't a developer command add permissions to this.

File: src/main/java/org/cascadebot/cascadebot/commands/management/FlagsCommand.java (hunk 6 above)
Comment (weeryan17) [inline src/main/java/org/cascadebot/cascadebot/commands/management/FlagsCommand.java]: Add a decription

**Comment by binaryoverload** [comment]
@JamesHawkinss Can we get the changes done here please? I need this merged to properly integrate into language

**Comment by JamesHawkinss** [comment]
@binaryoverload sorry for being late, all done now.

**Pull request merged**